More meaningful name for `@task.kubernetes` pods

insomnes opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->


<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

Use name based on python callable in `@task.kubernetes` pod name generation 

- generate specific pod name base by provided python callable
- drop uuid usage with no name (random is controlled by `random_name_suffix` arg, there is no need for extra UUID usage)
- add specific pod naming tests for decorator flow because decorator distinguishes between `name=None` and no `name` argument provided

closes: #46464

<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

insomnes committed:
Use name based on python callable in `@task.kubernetes` pod name generation

- generate specific pod name base by provided python callable
- drop uuid usage by default (random is controlled by `random_name_suffix` arg)
- add specific pod naming tests for decorator flow

insomnes committed:
Update decorator doc-string

RNHTTR reviewed (commented):
What will happen if two Dag runs are triggered at the same time, and two `TaskInstance`s are created from the same task simultaneously?

insomnes commented on airflow/decorators/__init__.pyi:
```diff
@@ -588,7 +588,7 @@ class TaskDecoratorCollection:
         :param name: Name of the pod to run. This will be used (plus a random
             suffix if *random_name_suffix* is *True*) to generate a pod ID
             (DNS-1123 subdomain, containing only ``[a-z0-9.-]``). Defaults to
```
I am sorry, but I don't get it.
What part of doc string should be updated or what should I add here? I believe current change is covered, what am I missing?

0BVer committed:
docs: clarify Gunicorn's role in webserver worker refresh (#46371)

chore: apply pre-commit

logarithm committed:
Allow passing empty labels in the spark kubernetes driver config (#45864)

* Allow passing empty labels in the driver config

* Fix formatting

* Fix formatting

---------

Co-authored-by: Maxim Logvinenko <maxim.logvinenko@coinmetrics.io>

bbossy committed:
SnowflakeSqlApiOperator snowflake_conn_id add to template_fields (#46422)

hussein-awala committed:
Replace Airflow email config by connection extras in SMTP provider (#46219)

* BREAKING CHANGE: replace Airflow config by conx extras in SMTP provider

* fix static checks

pierrejeambrun committed:
Integrate the SimpleAuthManager UI in dev mode (#46511)

pierrejeambrun committed:
AIP-84 Remove unecessary datamodels config and from_attributes (#46328)

jedcunningham committed:
Fix scheduler ServiceAccount automount for multi-executor (#46486)

We can't just check if CeleryExecutor is one of the executors when
determining if we can expose the automount param, we need to make sure
it's the _only_ executor - the others need it.

shubhamraj-git committed:
Add update XCom endpoint in RestAPI (#46457)

* Add XCom update API

* Add tests for XCom update API

* Fix failures on execution

sunank200 committed:
Fix missing logical_date in DagRun for DMS operator tests (#46521)

* Add logical in the tests

* Fix compact test

vincbeck committed:
Update AWS auth manager to use Fastapi instead of Flask (#46381)

andrii-korotkov-verkada committed:
Rename apiServer to _apiServer to preserve it in beta until v3 release (#46480)

Closes #46477

The chart version 1.16.0 would be released before airflow v3, so it's better to keep api server in beta so it's easier to make changes.

Signed-off-by: Andrii Korotkov <andrii.korotkov@verkada.com>

ashb committed:
Remove `ds`/`ts` etc from Template/run context when no logical date is defined (#46522)

This is part of the AIP-83 amendments.

feluelle committed:
Fix task sdk client dry-run mode (#46524)

* Fix task sdk client dry-run mode

The `run_after` field was missing in the fake dag run response.

* Fix trailing whitespaces

dabla committed:
Move microsoft winrm to new provider structure (#46469)

* refactor: Moved microsoft winrm provider

* refactor: fixed static checks

ellisms committed:
Adding extra links for EC2 (#46340)

* Update console link to filter on multiple instance ids

tirkarthi committed:
Add state filter to task page. (#46465)

* Add state filter to task page.

* Refactor task instance and dagrun states to constants util.

jason810496 committed:
Move CNCF Kubernetes to new provider structure (#46436)

* Move CNCF Kubernetes to new provider structure

* Fix doc include path and k8s test

* Fix taskflow tutorial

* Fix test_project_structure

* Strip src. prefix instead of replacing all src.

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

* Merge fix get_classes_from_file #46454

* Fix TestCncfProviderProjectStructure
- rename PROVIDER from "cncf" to "cncf/kubernetes"
- remove MISSING_EXAMPLES_FOR_CLASSES

* Fix k8s CI requirements

* fixup! Fix k8s CI requirements

---------

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

ashb committed:
Add dynamic task mapping into TaskSDK runtime (#46032)

The big change here (other than just moving code around) is to introduce a
conceptual separation between Definition/Execution time and Scheduler time.

This means that the expansion of tasks (creating the TaskInstance rows with
different map_index values) is now done on the scheduler, and we now
deserialize to different classes. For example, when we deserialize the
`DictOfListsExpandInput` it gets turned into an instance of
SchedulerDictOfListsExpandInput. This is primarily designed so that DB access
is kept 100% out of the TaskSDK.

Some of the changes here are on the "wat" side of the scale, and this is
mostly designed to not break 100% of our tests, and we have #45549 to look at
that more holistically.

To support the "reduce" style task which takes as input a sequence of all the
pushed (mapped) XCom values, and to keep the previous behaviour of not loading
all values in to memory at once, we have added a new HEAD route to the Task
Execution interface that returns the number of mapped XCom values so that it
is possible to implement `__len__` on the new LazyXComSequence class.

I have deleted a tranche of tests from tests/models that were to do with
runtime behavoir and and now tested in the TaskSDK instead.

bbovenzi committed:
Fix bug in query invalidation and remove custom predicate logic (#46485)

* Fix bug in query invalidation and remove custom predicate logic

* Address PR feedback

* Only use key fn when we have props to pass

jason810496 committed:
Move Apache Flink to new provider structure (#46132)

kunaljubce committed:
Revert "Revert "Added job_clusters as a templated parameter to CreateDatabric…" (#45066)

This reverts commit 09d8a803ca3f76d8598e7eeeb86397cf1cd4076d.

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

potiuk committed:
Move apache.impala provider to a new structure (#46532)

Part of #46045

kunaljubce committed:
Moving yandex provider to new provider structure (#46525)

* Moving yandex provider to new provider structure

* fixup! Moving yandex provider to new provider structure

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
